src/libraries/Microsoft.PowerFx.Core/Logging/Log.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using Microsoft.PowerFx.Core.Utils;
+
+namespace Microsoft.PowerFx.Logging
+{
+    /// <summary>
+    /// Provides a logging instance used by PowerFx. 
+    /// Default Logs to console in DEBUG, but hosts can register loggers to also collect PowerFx telemetry.
+    /// </summary>
+    public sealed partial class Log
+    {
+        private IPowerFxLogger _logger;
+
+        private static Log _instance;
+
+        internal static Log Instance => _instance ??= new Log();
+
+        private Log()
+        {
+#if DEBUG
+            _logger = new DebugLogger();
+#endif
+        }
+
+        internal void Track(string eventName, object customDimensions = null)
+        {
+            var serialized = customDimensions != null ? JsonSerializer.Serialize(customDimensions) : string.Empty;
+            _logger?.Track(eventName, serialized);
+        }
+
+        internal void Warning(string message)
+        {
+            _logger?.Warning(message);
+        }
+
+        internal void Exception(Exception exception, string message = "")
+        {
+            _logger?.Exception(exception, message);
+        }
+
+        internal void Error(string message)
+        {
+            _logger.Error(message);
+        }
+
+        internal LoggingScenario StartScenario(string scenarioName, object customDimensions = null)
+        {
+            var serialized = customDimensions != null ? JsonSerializer.Serialize(customDimensions) : string.Empty;
+            var scenario = new LoggingScenario(scenarioName);
+            _logger?.StartScenario(scenario.ScenarioName, scenario.ScenarioInstance, serialized);
+            return scenario;
+        }
+
+        internal void EndScenario(LoggingScenario scenario, object customDimensions = null)
+        {
+            scenario.CloseScenario();
+            var serialized = customDimensions != null ? JsonSerializer.Serialize(customDimensions) : string.Empty;
+            _logger?.EndScenario(scenario.ScenarioName, scenario.ScenarioInstance, serialized);
```

Comment:
We probably should enforce some basic schema (Not sure how) for logged content. 

For example. 
timestamp
session id/request id, 
operation Id
etc.

Reply:
IMO, those are things that probably should get added by the host in their implementation of IPowerFxLogger. We don't really know much about what context we're running in within the Power Fx code, and every host might have different concepts of sessions/operations. 

For MSFT hosts, we can work with them on that, for others, it depends on if/how the hosts would want to make use of that telemetry.

Reply:
We should discuss how we want to track the usage. Having a consistent schema for MS internal teams will make it easy to track usage.

Reply:
+1 to both points.  But I think we need to talk with the hosts first to make sure there's a e2e plan here - including that we can get meaningful data from wherever they're storing it.